include/cantera/numerics/Newton.h

```diff
@@ -0,0 +1,100 @@
+//! @file Newton.h
```

Comment:
Based on `MultiNewton`, adds solution component tolerance/bounds functionality from `Domain1D`, and internal storage for a `FuncEval` residual class and a Jacobian.